src/main/java/listeners/MainEventListener.java

```diff
@@ -0,0 +1,232 @@
+        System.out.println("Logged in");
+        event.getJDA().getPresence().setPresence(Activity.playing("$help"), false);
+    }
+
+    @Override
+    public void onGuildMessageReceived(@NotNull GuildMessageReceivedEvent event) {
+        String[] tokens = event.getMessage().getContentDisplay().split(" ");
+        if (!tokens[0].startsWith("$")) return;
+
+        String guildID = event.getGuild().getId();
+        Server server = servers.computeIfAbsent(guildID, k -> Server.load(event.getGuild()));
+
+        // pick the correct method to call
+        CommandHandler commandHandler;
+        switch (tokens[0].substring(1)) {
+            case "help" -> commandHandler = this::help;
+            case "maketopic" -> commandHandler = this::maketopic;
+            case "deletetopic" -> commandHandler = this::deletetopic;
+            case "showtopics" -> commandHandler = this::showtopics;
+            case "queue" -> commandHandler = this::queue;
+            case "ready" -> commandHandler = this::ready;
+            case "showqueue" -> commandHandler = this::showqueue;
+            case "clear" -> commandHandler = this::clear;
+            default -> commandHandler = this::unknownCommand;
+        }
+        commandHandler.handle(
+                Objects.requireNonNull(event.getMember()),
+                event.getChannel(),
+                server,
+                Arrays.copyOfRange(tokens, 1, tokens.length));
+    }
+
+    private void help(Member member, TextChannel channel, Server server, String[] args) {
+        EmbedBuilder embedBuilder = new EmbedBuilder();
+        embedBuilder.setTitle("Help!");
+        embedBuilder.setDescription("Possible commands:");
+        embedBuilder.setColor(0xE57D25);
+
+        embedBuilder.addField("$queue <topic>", "Add yourself to a queue.", false);
+        embedBuilder.addField("$showqueue <topic>", "Show the people currently in queue.", false);
+        embedBuilder.addField("$showtopics", "List all topics.", false);
+        embedBuilder.addField("$ready <topic> (admin only)", "Retrieve the next person from the queue.", false);
+        embedBuilder.addField("$clear <topic> (admin only)", "Clear the specified queue.", false);
+        embedBuilder.addField("$maketopic <name> (admin only)", "Create a new topic.", false);
+        embedBuilder.addField("$deletetopic <name> (admin only)", "Delete a topic.", false);
+
+        channel.sendMessage(embedBuilder.build()).queue();
+    }
+
+    private void maketopic(Member member, TextChannel channel, Server server, String[] args) {
+        if (!checkAdmin(member, channel)) return;
+
+        Topic topic = new Topic(args[0]);
+        server.createTopic(topic);
+
+        channel.sendMessage(String.format(
+                "%s Topic role \"%s\" has been created.",
+                member.getAsMention(),
+                args[0])).queue();
+    }
+
+    private void deletetopic(Member member, TextChannel channel, Server server, String[] args) {
```

Comment:
camelCase?

Reply:
I was trying to name the per-command functions to be identical to how they are represented to the user. In this case, the actual command is `$deletetopic <topic>`, so I wanted the function name to match. However, I suppose that it doesn't matter since the user can't see the code anyway, so I can swap these to camelCase.